pkg/kubelet/container/runtime.go

```diff
@@ -381,14 +381,23 @@ type PortMapping struct {
 	HostIP string
 }
 
+type DeviceInfo struct {
+	// Path on host for mapping
+	PathOnHost string
+	// Path in Container to map
+	PathInContainer string
+	// Cgroup permissions
+	Permissions string
+}
+
 // RunContainerOptions specify the options which are necessary for running containers
 type RunContainerOptions struct {
 	// The environment variables list.
 	Envs []EnvVar
 	// The mounts for the containers.
 	Mounts []Mount
 	// The host devices mapped into the containers.
-	Devices []string
+	Devices []DeviceInfo
```

Comment:
So far, it doesn't need api.PodSpec change for this. This is a general devices mapping, GPU is an example of these devices, but we can consider it when enable GPU in kubernetes.  Regarding the implementation, you can see it in #28216 , also there is a PR for GPU, #30756